server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -340,8 +342,9 @@ private void snapshot(final ShardId shardId, final Snapshot snapshot, final Inde
             try {
                 // we flush first to make sure we get the latest writes snapshotted
                 snapshotRef = indexShard.acquireLastIndexCommit(true);
+                final IndexCommit indexCommit = snapshotRef.getIndexCommit();
```

Comment:
I'm not sure 100% I like computing this here and passing it as yet another parameter to `repository.snapshotShard`.
In hindsight, the API here would be a lot easier/flexible to deal with had we not removed the `IndexShard` in https://github.com/elastic/elasticsearch/pull/42213 ... maybe in a follow up we could introduce some object that exposes the shard related things like `Store` etc. to simplify this API.